dojo-dungeon/src/dojo/compiler/DojoCompiler.java

```diff
@@ -31,81 +22,28 @@ public class DojoCompiler {
    */
   public record TestResult(String testName, boolean passed, List<String> messages) {}
 
+  private static final String ABSOLUTE_BUILD_PATH =
+      System.getProperty("dojoDungeonAbsBuildDir", "build/classes/java/main");
```

Comment:
Die Variable heisst "**ABSOLUTE**_BUILD_PATH". Wenn das über die System-Property gesetzt wird, dann ist es auch ein absoluter Pfad. Aber wenn das nicht klappt, dann hast Du nur einen relativen Pfad.

(a) Hast Du mal geprüft, ob der Code (`DojoCompiler`) dann auch noch funktioniert, also mit dem relativen Pfad?
(b) Sollte die Variable besser anders heissen?
(c) Wäre es vielleicht besser, hier eine Exception zu werfen, das Fehlen der System-Property zu loggen und das Programm zu beenden?

Reply:
Hmmm. 

Wenn es tatsächlich auch mit dem relativen Fallback-Pfad klappt, dann sollte die Variable wohl wirklich eher `BUILD_PATH` heißen (auch wenn von Gradle im Erfolgsfall ein absoluter Pfad geliefert wird).

`static` macht auch sehr viel Sinn, weil das ja nur einmal bestimmt werden muss ...

Exception: Ich bin hier wirklich unsicher, was wohl am sinnvollsten sein könnte. (A) Wenn man mit dem Fallback-Pfad arbeitet, dann läuft das Spiel auch ohne Gradle durch (sofern man den Rest irgendwie konfiguriert kriegt) - dann gibt es eigentlich auch keinen Grund, eine Exception zu werfen und auszusteigen. (B) Auf der anderen Seite finde ich diese hart kodierten Fallback-Pfade immer unschön. Man weiss weder als User noch als Programmier davon, und wenn dann mal was nicht funktioniert, sucht man sich einen Wolf. Da ist es imho ehrlicher, wenn man das Programm einfach abschießt, wenn dieser notwendige Pfad von Gradle nicht gesetzt wurde (außerdem dürfte dann sowieso einiges anderes noch ziemlich kaputt sein). _TDLR_: Ich denke, wir sollten auf den hart kodierten Fallback-Pfad verzichten und stattdessen bei Fehlen der System-Property tatsächlich direkt eine Exception werfen. Vermutlich muss die Initialisierung der statischen Variable dazu dann in einem static-Block erfolgen.

Reply:
Das sollte ich eigentlich bei dem analogen Fall beim Logging auch so machen - dort ist im Moment auch ein Fallback-Pfad hart kodiert, was im obigen Sinne nicht wirklich schön ist. Wenn Du hier mit einer Exception arbeitest, dann sollte das beim Logging-Ordner auch so gemacht werden.

Reply:
@cagix So ungefähr wie in [696b24c](https://github.com/Dungeon-CampusMinden/Dungeon/pull/1532/commits/696b24ce8e4f8c6b9b626a5f31298ab3f46e2af9)? Das sieht nun erst mal etwas "holprig" aus finde ich - aber anders funktioniert es glaube ich nicht.